src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1740,23 +1742,29 @@ App::DocumentObjectExecReturn* Hole::execute()
             double holeCutDepth = HoleCutDepth.getValue();
             double countersinkAngle = Base::toRadians(HoleCutCountersinkAngle.getValue() / 2.0);
 
-            if ( isCounterbore) {
-                // Counterbore is rendered the same way as a countersink, but with a hardcoded angle of 90deg
+            if (isCounterbore) {
+                // Counterbore is rendered the same way as a countersink, but with a hardcoded
+                // angle of 90deg
                 countersinkAngle = Base::toRadians(90.0);
             }
 
-            if ( isCountersink) {
-                holeCutDepth = 0;
+            if (isCountersink) {
+                holeCutDepth = 0.0;
+                // We cannot recalculate the HoleCutDiameter because the previous HoleCutDepth
+                // is unknown. Therefore we cannot know with what HoleCutDepth the current
+                // HoleCutDiameter was calculated.

```

Comment:
shouldn't we also reset the holeCutDepth property then?

Reply:
You mean we should block to set the depth? This would indeed be sensible but let me think about this further